radiation/python/radlw/taumol_gt4py.py

```diff
@@ -0,0 +1,4957 @@
+            )
+
+            taug[0, 0, 0][ig2] = corradj * (
+                colamt[0, 0, 0][0]
+                * (
+                    fac00 * absb[0, 0, 0][ig2, ind0]
+                    + fac10 * absb[0, 0, 0][ig2, ind0p]
+                    + fac01 * absb[0, 0, 0][ig2, ind1]
+                    + fac11 * absb[0, 0, 0][ig2, ind1p]
+                )
+                + taufor
+                + taun2
+            )
+
+            fracs[0, 0, 0][ig2] = fracrefb[0, 0, 0][ig2]
+
+
+@stencil(
+    backend=backend,
+    rebuild=rebuild,
+    externals={
+        "nspa": nspa[1],
+        "nspb": nspb[1],
+        "laytrop": indict["laytrop"],
+        "ng02": ng02,
+        "ns02": ns02,
+        "nlay": nlay,
+    },
+)
+def taugb02(
+    pavel: FIELD_FLT,
+    colamt: Field[type_maxgas],
+    fac00: FIELD_FLT,
+    fac01: FIELD_FLT,
+    fac10: FIELD_FLT,
+    fac11: FIELD_FLT,
+    jp: FIELD_INT,
+    jt: FIELD_INT,
+    jt1: FIELD_INT,
+    selffac: FIELD_FLT,
+    selffrac: FIELD_FLT,
+    indself: FIELD_INT,
+    forfac: FIELD_FLT,
+    forfrac: FIELD_FLT,
+    indfor: FIELD_INT,
+    fracs: Field[type_ngptlw],
+    taug: Field[type_ngptlw],
+    absa: Field[(DTYPE_FLT, (10, 65))],
+    absb: Field[(DTYPE_FLT, (10, 235))],
+    selfref: Field[(DTYPE_FLT, (10, 10))],
+    forref: Field[(DTYPE_FLT, (10, 4))],
+    fracrefa: Field[(DTYPE_FLT, (10,))],
+    fracrefb: Field[(DTYPE_FLT, (10,))],
+    ind0: FIELD_INT,
+    ind0p: FIELD_INT,
+    ind1: FIELD_INT,
+    ind1p: FIELD_INT,
+    inds: FIELD_INT,
+    indsp: FIELD_INT,
+    indf: FIELD_INT,
+    indfp: FIELD_INT,
+    corradj: FIELD_FLT,
+    tauself: FIELD_FLT,
+    taufor: FIELD_FLT,
+):
+    from __externals__ import nspa, nspb, laytrop, ng02, nlay, ns02
+
+    with computation(PARALLEL), interval(0, laytrop):
+        ind0 = ((jp - 1) * 5 + (jt - 1)) * nspa
+        ind1 = (jp * 5 + (jt1 - 1)) * nspa
+        inds = indself - 1
+        indf = indfor - 1
+
+        ind0p = ind0 + 1
+        ind1p = ind1 + 1
+        indsp = inds + 1
+        indfp = indf + 1
+
+        corradj = 1.0 - 0.05 * (pavel - 100.0) / 900.0
+
+        for ig in range(ng02):
+            tauself = selffac * (
+                selfref[0, 0, 0][ig, inds]
+                + selffrac * (selfref[0, 0, 0][ig, indsp] - selfref[0, 0, 0][ig, inds])
+            )
+            taufor = forfac * (
+                forref[0, 0, 0][ig, indf]
+                + forfrac * (forref[0, 0, 0][ig, indfp] - forref[0, 0, 0][ig, indf])
+            )
+
+            taug[0, 0, 0][ns02 + ig] = corradj * (
+                colamt[0, 0, 0][0]
+                * (
+                    fac00 * absa[0, 0, 0][ig, ind0]
+                    + fac10 * absa[0, 0, 0][ig, ind0p]
+                    + fac01 * absa[0, 0, 0][ig, ind1]
+                    + fac11 * absa[0, 0, 0][ig, ind1p]
+                )
+                + tauself
+                + taufor
+            )
```

Comment:
man functions sure would be nice for these repeated patterns, but there are known issues with them, so good to leave for now.

Reply:
agreed